runtime/metrics.go

```diff
@@ -1,171 +1,177 @@
-	default:
+func (m *Metrics) logSinkJSON(filename string) MetricSinkFn {
+	f, err := m.re.CreateRawAsset(filename)
+	if err != nil {
 		panic(err)
 	}
-	return m
+
+	enc := json.NewEncoder(f)
+	return func(m *Metric) error {
+		return enc.Encode(m)
+	}
 }
 
-func (*Metrics) NewSummaryVec(o SummaryOpts, labels ...string) *SummaryVec {
-	m := prometheus.NewSummaryVec(o, labels)
-	switch err := prometheus.Register(m); err.(type) {
-	case nil, prometheus.AlreadyRegisteredError:
-	default:
-		panic(err)
+func (m *Metrics) writeToInfluxDBSink(measurement string) MetricSinkFn {
+	return func(metric *Metric) error {
+		fields := make(map[string]interface{}, len(metric.Measures))
+		for k, v := range metric.Measures {
+			key := strings.Join([]string{metric.Name, metric.Type.String(), k}, ".")
+			fields[key] = v
+		}
+
+		p, err := client.NewPoint(measurement, m.tags, fields, time.Unix(0, metric.Timestamp))
+		if err != nil {
+			return err
+		}
+		m.batcher.WritePoint(p)
+		return nil
 	}
-	return m
 }
 
-// HTTPPeriodicSnapshots periodically fetches the snapshots from the given address
-// and outputs them to the out directory. Every file will be in the format timestamp.out.
-func (re *RunEnv) HTTPPeriodicSnapshots(ctx context.Context, addr string, dur time.Duration, outDir string) error {
-	err := os.MkdirAll(path.Join(re.TestOutputsPath, outDir), 0777)
-	if err != nil {
-		return err
+func (m *Metrics) recordEvent(evt *Event) {
+	if m.influxdb == nil {
+		return
 	}
 
-	nextFile := func() (*os.File, error) {
-		timestamp := strconv.FormatInt(time.Now().Unix(), 10)
-		return os.Create(path.Join(re.TestOutputsPath, outDir, timestamp+".out"))
+	// this map copy is terrible; the influxdb v2 SDK makes points mutable.
+	tags := make(map[string]string, len(m.tags)+1)
+	for k, v := range m.tags {
+		tags[k] = v
 	}
 
-	go func() {
-		ticker := time.NewTicker(dur)
-		defer ticker.Stop()
-
-		for {
-			select {
-			case <-ctx.Done():
-				return
-			case <-ticker.C:
-				func() {
-					req, err := http.NewRequestWithContext(ctx, "GET", addr, nil)
-					if err != nil {
-						re.RecordMessage("error while creating http request: %v", err)
-						return
-					}
-
-					resp, err := http.DefaultClient.Do(req)
-					if err != nil {
-						re.RecordMessage("error while scraping http endpoint: %v", err)
-						return
-					}
-					defer resp.Body.Close()
-
-					file, err := nextFile()
-					if err != nil {
-						re.RecordMessage("error while getting metrics output file: %v", err)
-						return
-					}
-					defer file.Close()
-
-					_, err = io.Copy(file, resp.Body)
-					if err != nil {
-						re.RecordMessage("error while copying data to file: %v", err)
-						return
-					}
-				}()
-			}
-		}
-	}()
+	tags["type"] = string(evt.Type)
 
-	return nil
+	if evt.Outcome != "" {
+		tags["outcome"] = string(evt.Outcome)
+	}
+
+	f := map[string]interface{}{
+		"error": evt.Error,
+	}
+
+	p, err := client.NewPoint("events", tags, f)
```

Comment:
Same comment as above - I think the schema here should be `events.%s` where %s is the event name. Then every event generally has its own set of fields and tags that are filled for every single entry.